test/test_commands_serve.rb

```diff
@@ -19,6 +19,11 @@ def custom_opts(what)
           p
         )
       end
+      allow(SafeYAML).to receive(:load_file).and_return({})
+      allow(Jekyll::Commands::Build).to receive(:build).and_return("")
+    end
+    teardown do
+      Jekyll.instance_variable_set(:@sites, [])
```

Comment:
You should be able to do [`Jekyll.sites.clear`](https://docs.ruby-lang.org/en/2.0.0/Array.html#method-i-clear) here.